cms/djangoapps/contentstore/utils.py

```diff
@@ -754,3 +774,165 @@ def get_course_grading(course_key):
+                entrance_exam_minimum_score_pct = entrance_exam_minimum_score_pct / 100
+            # If there's already an entrance exam defined, we'll update the existing one
+            if course_entrance_exam_present:
+                exam_data = {
+                    'entrance_exam_minimum_score_pct': entrance_exam_minimum_score_pct
+                }
+                update_entrance_exam(request, course_key, exam_data)
+            # If there's no entrance exam defined, we'll create a new one
+            else:
+                create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct)
+
+        # If the entrance exam box on the settings screen has been unchecked,
+        # and the course has an entrance exam attached...
+        elif not entrance_exam_enabled and course_entrance_exam_present:
+            delete_entrance_exam(request, course_key)
+
+    # Perform the normal update workflow for the CourseDetails model
+    return CourseDetails.update_from_json(course_key, payload, request.user)
+
+
+def get_course_settings(request, course_key, course_block):
+    """
+    Utils is used to get context of course settings.
+    It is used for both DRF and django views.
+    """
+
+    from .views.course import get_courses_accessible_to_user, _process_courses_list
+
+    credit_eligibility_enabled = settings.FEATURES.get('ENABLE_CREDIT_ELIGIBILITY', False)
+    upload_asset_url = reverse_course_url('assets_handler', course_key)
+
+    # see if the ORG of this course can be attributed to a defined configuration . In that case, the
+    # course about page should be editable in Studio
+    publisher_enabled = configuration_helpers.get_value_for_org(
+        course_block.location.org,
+        'ENABLE_PUBLISHER',
+        settings.FEATURES.get('ENABLE_PUBLISHER', False)
+    )
+    marketing_enabled = configuration_helpers.get_value_for_org(
+        course_block.location.org,
+        'ENABLE_MKTG_SITE',
+        settings.FEATURES.get('ENABLE_MKTG_SITE', False)
+    )
+    enable_extended_course_details = configuration_helpers.get_value_for_org(
+        course_block.location.org,
+        'ENABLE_EXTENDED_COURSE_DETAILS',
+        settings.FEATURES.get('ENABLE_EXTENDED_COURSE_DETAILS', False)
+    )
+
+    about_page_editable = not publisher_enabled
+    enrollment_end_editable = GlobalStaff().has_user(request.user) or not publisher_enabled
+    short_description_editable = configuration_helpers.get_value_for_org(
+        course_block.location.org,
+        'EDITABLE_SHORT_DESCRIPTION',
+        settings.FEATURES.get('EDITABLE_SHORT_DESCRIPTION', True)
+    )
+    sidebar_html_enabled = ENABLE_COURSE_ABOUT_SIDEBAR_HTML.is_enabled()
+
+    verified_mode = CourseMode.verified_mode_for_course(course_key, include_expired=True)
+    upgrade_deadline = (verified_mode and verified_mode.expiration_datetime and
+                        verified_mode.expiration_datetime.isoformat())
+    settings_context = {
```

Comment:
`date_placeholder_format` [recently introduced](https://github.com/open-craft/edx-platform/blob/opencraft-release/palm.1/cms/djangoapps/contentstore/views/course.py#L1230) is missing here. This leads to exception when trying to open the course settings page in studio.

Reply:
Thanks for catching that! I've updated it now.